src/movies/movies.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, Like } from 'typeorm';
+import { Movie } from '../entities/Movie/movie.entity';
+import { MovieDto } from '../entities/Movie/dto/movie.dto';
+import { UUID } from 'crypto';
+
+@Injectable()
+export class MovieService {
+  constructor(
+    @InjectRepository(Movie)
+    private readonly movieRepository: Repository<Movie>,
+  ) {}
+
+  async findMovies(page: number, limit: number, sort: string, search: string, genre: string): Promise<{ movies: MovieDto[], total: number }> {
+    try {
```

Comment:
The try catch needs to be beneficial either by logging some information related to the error and its params or perhaps doing a different action, but as it stands it's not really adding much